Formula/libsql-server.rb

```diff
@@ -0,0 +1,28 @@
+class LibsqlServer < Formula
+  desc "Server-mode binary for libSQL (sqld), an Open Contribution fork of SQLite"
+  homepage "https://libsql.org"
+  url "https://github.com/libsql/sqld/archive/refs/tags/v0.9.0.tar.gz"
+  sha256 "d3b222ed54c4ab1707da16198760f8619d5cdbfbc6ecd72998671100ddc85439"
+  license "MIT"
+
+  depends_on "llvm" => :build
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+  depends_on "openssl@3"
+  depends_on "protobuf"
+
+  def install
+    system "cargo", "install", *std_cargo_args(path: "sqld")
```

Comment:
Let's do this then to make sure it's not statically linking to its own OpenSSL:
```suggestion
    ENV["OPENSSL_NO_VENDOR"] = "1"
    ENV["OPENSSL_DIR"] = Formula["openssl@3"].opt_prefix
    system "cargo", "install", *std_cargo_args(path: "sqld")
```

Reply:
Still not shown:
```
/opt/homebrew/bin//sqld:
	/System/Library/Frameworks/Security.framework/Versions/A/Security (compatibility version 1.0.0, current version 60158.100.133)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1858.112.0)
	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
```

So I investigated it a bit more, and the tls dependency is coming from the `tonic` crate, which itself pulls `tokio-rustls`. `rustls` is a full rust implementation, so it's not supposed to pull in a dependency to openssl.

What surprises me is why is Linux doing it. I'll try to build it on Linux to understand more. Thanks for all the help so far!

Reply:
> What surprises me is why is Linux doing it.

It's probably enough to do `cargo tree` inside the source directory on Linux. I definitely saw an `openssl-sys` dependency there -- I just didn't track down where it came from.

Reply:
Ok. What's your suggestion here, @carlocab ? We found out the culprit, that seems to be the `localtunnel` crate, and we're trying to patch it upstream: https://github.com/MarinPostma/rlt/pull/1, but at this point it is a dependency we don't control, we don't know how responsive they would be, etc.

We would love to add our package to homebrew before resolving that =(

Reply:
If it's vendoring it's own OpenSSL that's a pretty clear case of https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae

Reply:
It's not vendoring openssl, it is linking against openssl.
The thing is that it doesn't do it on mac (where it can use rustls), but does it on Linux. We have a pretty good understanding of why this is happening now, and working with the crate author to resolve this long term.

But for now I just pushed a version that adds the openssl dependency on Linux only if that's acceptable to y'all

Reply:
I suggest adding `reqwest` to your `Cargo.toml` so that you have direct control over the requested features without being blocked by upstream.

Also, I think we'd rather link to OpenSSL rather than to rustls, even on macOS.

Reply:
fwiw, we removed the dependency that was causing issues, and this shouldn't be an issue anymore. I just pushed an update with 0.10 that removes the functionality that was bringing in the problematic dependency